mkosi/__init__.py

```diff
@@ -1292,14 +1292,31 @@ def gzip_binary(context: Context) -> str:
     return "pigz" if context.config.find_binary("pigz") else "gzip"
 
 
+def kernel_get_ver_from_release(context: Context) -> str:
+    # Try to get version from the first dir under usr/lib/modules but fail if multiple versions are found
+    versions = [p.name for p in (context.root / "usr/lib/modules").glob("[0-9]*")]
+    if len(versions) > 1:
+        die(
+            "Multiple kernel module directories found in /usr/lib/modules, unable to determine correct version to use"
+        )
+    elif len(versions) == 0 or not re.match(r"\d+\.\d+.*", versions[0]):
```

Comment:
Could this become
```suggestion
    versions = [p.name for p in (context.root / "usr/lib/modules").glob("*") if re.match(r"\d+\.\d+.*", p.name)]
    if len(versions) > 1:
        die(
            "Multiple kernel module directories found in /usr/lib/modules, unable to determine correct version to use"
        )
    elif len(versions) == 0:
```
This way it is guaranteed that if for some reason there would be more than a single thing matching the glob, but only one matching the regex, the right one would be picked instead of the erroring out.